test/Semantics/select-rank02.f90

```diff
@@ -0,0 +1,66 @@
+! RUN: %S/test_errors.sh %s %flang %t
+!Shape analysis related tests for SELECT RANK Construct(R1148)
+program select_rank
+   implicit none
+   integer, dimension(2,3):: arr_pass
+   call check(arr_pass)
+
+contains
+    subroutine check(arr)
+        implicit none
+        integer :: arr(..)
+        INTEGER :: j
+        select rank (arr)
+            rank(2)
+                j = INT(0, KIND=MERGE(KIND(0), -1, RANK(arr) == SIZE(SHAPE(arr)))) !arr is dummy
+        end select
+    end subroutine
+    subroutine check2(arr)
+        implicit none
+        integer :: arr(..)
+        INTEGER :: j
+        integer,dimension(-1:10, 20:30) :: brr
+
+        select rank (arr)
+            rank(2)
+                j = INT(0, KIND=MERGE(KIND(0), -1, RANK(arr) == SIZE(SHAPE(brr)))) !brr is local to subroutine
```

Comment:
I don't understand what's being tested here.

Reply:
Ok, I was trying to test with a local copy and the dummy arguments.
Remove this test?